superset-frontend/src/SqlLab/components/ExploreCtasResultsButton/index.jsx

```diff
@@ -35,29 +35,23 @@ const propTypes = {
   templateParams: PropTypes.string,
 };
 
-class ExploreCtasResultsButton extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.visualize = this.visualize.bind(this);
-    this.onClick = this.onClick.bind(this);
-  }
-
-  onClick() {
-    this.visualize();
-  }
-
-  buildVizOptions() {
-    return {
-      datasourceName: this.props.table,
-      schema: this.props.schema,
-      dbId: this.props.dbId,
-      templateParams: this.props.templateParams,
-    };
-  }
+const ExploreCtasResultsButton = (
+  table,
+  schema,
+  dbId,
+  templateParams,
+  errorMessage,
+  { createCtasDatasource, addInfoToast, addDangerToast } = actions,
+) => {
```

Comment:
in order to convert this from props (one argument), shouldn't this be a  singular destructured object instead of positional arguments?